From 361bd2430c44693db8978e4004ed355f4b5f1bab Mon Sep 17 00:00:00 2001
From: Peter Naulls <peter@chocky.org>
Date: Thu, 25 Aug 2011 22:07:14 +0000
Subject: [PATCH] authsae o11s authentication daemon

This adds the o11s cozybit authentication daemon fork of authsae (see o11s.org).
  This requires forthcoming work on o11s in the
kernel/compat-wireless et al. but there are people interested in running
this under OpenWrt right now.

I have not added any init scripts, etc.

Signed-off-by: Peter Naulls <peter@chocky.org>
---
 package/net/authsae/Makefile |   53 ++++++++++++++++++++++++++++++++++++++++++
 package/openssl/Makefile     |    6 ++++-
 2 files changed, 58 insertions(+), 1 deletions(-)
 create mode 100644 package/net/authsae/Makefile

diff --git a/package/net/authsae/Makefile b/package/net/authsae/Makefile
new file mode 100644
index 0000000..4da1198
--- /dev/null
+++ b/package/net/authsae/Makefile
@@ -0,0 +1,53 @@
+
+# Copyright (C) 2007-2010 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=authsae
+PKG_VERSION:=0.01
+PKG_RELEASE:=1
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=git://github.com/cozybit/authsae.git
+PKG_REV:=HEAD
+PKG_SOURCE_VERSION:=$(PKG_REV)
+
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/authsae
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=wireless
+  TITLE:=o11s mesh security
+  DEPENDS=+libopenssl +libconfig +libnl
+endef
+
+TARGET_CFLAGS += -D_GNU_SOURCE
+
+define Package/authsae/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/linux/mon \
+                       $(PKG_BUILD_DIR)/linux/meshd \
+                       $(PKG_BUILD_DIR)/linux/meshd-nl80211 $(1)/usr/bin/
+	$(INSTALL_DIR) $(1)/etc
+	$(INSTALL_CONF) $(PKG_BUILD_DIR)/config/authsae.sample.cfg $(1)/etc/authsae.cfg
+endef
+
+define Package/conffiles
+	/etc/authsae.cfg
+endef
+
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR)/linux $(TARGET_CONFIGURE_OPTS)
+endef
+
+$(eval $(call BuildPackage,authsae))
diff --git a/package/openssl/Makefile b/package/openssl/Makefile
index b9b9b31..8bbe2e9 100644
--- a/package/openssl/Makefile
+++ b/package/openssl/Makefile
@@ -74,7 +74,7 @@ endef
 
 OPENSSL_NO_CIPHERS:= no-idea no-md2 no-mdc2 no-rc5 no-sha0 no-smime \
 					no-rmd160 no-aes192 no-ripemd no-camellia no-ans1 no-krb5
-OPENSSL_OPTIONS:= shared no-ec no-err no-hw no-threads zlib-dynamic no-sse2
+OPENSSL_OPTIONS:= shared no-err no-hw no-threads zlib-dynamic no-sse2
 
 ifdef CONFIG_OPENSSL_ENGINE
   OPENSSL_OPTIONS += -DHAVE_CRYPTODEV
@@ -82,6 +82,10 @@ else
   OPENSSL_OPTIONS += no-engines
 endif
 
+ifndef CONFIG_PACKAGE_authsae
+  OPENSSL_OPTIONS += no-ec
+endif
+
 OPENSSL_OPTIONS += no-perlasm
 
 define Build/Configure
-- 
1.7.1

